src/Context.php

```diff
@@ -119,6 +119,10 @@ public function addSymbol($name)
      */
     public function addVariable(Variable $variable)
     {
+        if (!is_string($variable->getName())) {
```

Comment:
It's one variable, not variable in variable, should be resolved by `Compiler` component
I dont see any pluses to use Variable in Variable :)